components/miscShared/fn_insertLoadoutIntoCrate.sqf

```diff
@@ -11,44 +11,21 @@ clearItemCargoGlobal _crate;
 clearBackpackCargoGlobal _crate;
 
 
-_backpacks = [];
-_weapons = [];
-_magazines = [];
-_items = [];
+private _backpacks = [];
+private _weapons = [];
+private _magazines = [];
+private _items = [];
 
 
-_appendWeapon =
+private _appendWeapon =
 {
     params ["_weapons", "_magazines", "_items", "_array"];
 
-    _weapon = _array select 0;
-    _muzzleDevice = _array select 1;
-    _pointer = _array select 2;
-    _optic = _array select 3;
-    _primaryMag = _array select 4;
-
-    _secondaryMag = [];
-    _bipod = "";
-
-    if (count _array < 7) then
-    {
-        _bipod = _array select 5;
-    }
-    else
-    {
-        _secondaryMag = _array select 5;
-        _bipod = _array select 6;
-    };
-
-    _weapons pushBack [_weapon, 1];
-    if (count _primaryMag > 0) then {_magazines pushBack _primaryMag};
-    if (count _secondaryMag > 0) then {_magazines pushBack _secondaryMag};
-    _items append ([_muzzleDevice, _pointer, _optic, _bipod] call BIS_fnc_consolidateArray);
-
+    _weapons pushBack _array;
 };
```

Comment:
Perhaps we could get rid of this function? We currently still call it with the full 4 arguments, despite only needing 2 in order to do ` _weapons pushBack _array`. Instead we could simply replace the function calls with that line - improves readability, and reduces the amount of script calls.

Reply:
Done!